cardano-api/src/Cardano/Api/IO.hs

```diff
@@ -76,7 +76,7 @@ newtype File content (direction :: FileDirection) = File
 handleFileForWritingWithOwnerPermission
   :: FilePath
   -> (Handle -> IO ())
-  -> IO (Either (FileError ()) ())
```

Comment:
This can't be `()` because we wan't to use the functions for reading things with `e` other than `()`.

The fact that `e` is here at all is problematic.   `e` is for decode errors, but why should functions purely used for reading and writing files and do no decode at all have to concern themselves with decode errors?  Having the `e` here produce type-inference problems down the line.